tests/python/contrib/test_ethosu/infra.py

```diff
@@ -42,6 +58,218 @@ class AttachType(IntEnum):
+                raise RuntimeError("ValueError :: tvm primary ifm base should be zero")
+            continue
+
+        # OFM regions should be cached to be checked later
+        if vela_cmd == cmd0.NPU_SET_OFM_REGION:
+            ofm_region_params.append((vela_value, tvm_value))
+            continue
+
+        # OFM bases should be cached to be checked later
+        if vela_cmd == cmd1.NPU_SET_OFM_BASE0:
+            ofm_bases.append((vela_value, tvm_value))
+            continue
+
+        check_cmms_equivalency(vela_cmd, vela_value, tvm_value, ignore_commands)
+
+    # The last OFM region could be different but it should be 1 and 4.
+    last_vela_ofm_region, last_tvm_ofm_region = ofm_region_params.pop(-1)
+    if not (last_vela_ofm_region == 1 and last_tvm_ofm_region == 4):
+        raise RuntimeError(
+            "ValueMismatch :: vela={}, tvm={} for last ofm region it should be 1 and 4 respectively".format(
+                last_vela_ofm_region, last_tvm_ofm_region
+            )
+        )
+
+    # The rest of the OFM regions should be the same.
+    for vela_value, tvm_value in ofm_region_params:
+        check_cmms_equivalency(vela_cmd, vela_value, tvm_value, ignore_commands)
+
+    # The last OFM base should be zero for tvm
+    _, last_tvm_ofm_base = ofm_bases.pop(-1)
+    if not last_tvm_ofm_base == 0:
+        raise RuntimeError("ValueError :: tvm primary ofm base should be zero")
+
+
+def deserialize_command_stream(blob):
+    assert isinstance(blob, bytes)
+    payload_bytes = struct.unpack("<{0}I".format(len(blob) // 4), blob)
+    cmms = []
+    # remove_header
+    payload_bytes = payload_bytes[8:]
+    idx = 0
+    while idx < len(payload_bytes):
+        cmd = []
+        code = payload_bytes[idx]
+        idx += 1
+        cmd.append(code)
+        payload_mode = CmdMode(code & CmdMode.Mask)
+        if payload_mode == CmdMode.Payload32:
+            value = payload_bytes[idx]
+            idx += 1
+            cmd.append(value)
+        cmms.append(cmd)
+    return cmms
+
+
+def _create_test_runner(accel):
+    file_dir = os.path.dirname(os.path.abspath(__file__))
+    test_root = os.path.join(file_dir, "reference_system")
+    ethosu_macs = accel[accel.rfind("-") + 1 :]
+    return AOTTestRunner(
+        makefile="corstone300",
+        prologue="""
+        uart_init();
+        EthosuInit();
+        """,
+        includes=["uart.h", "ethosu_55.h", "ethosu_mod.h", "hard_fault.h"],
+        parameters={"ETHOSU_TEST_ROOT": test_root, "NPU_VARIANT": ethosu_macs},
+        pass_config={
+            "relay.ext.ethosu.options": {
+                "accelerator_config": accel,
+            }
+        },
+    )
+
+
+def build_source(module, inputs, outputs, accel="ethos-u55-256"):
+    test_runner = _create_test_runner(accel)
+    return compile_models(
+        models=AOTTestModel(
+            module=module,
+            inputs=inputs,
+            outputs=outputs,
+            output_tolerance=10,
```

Comment:
I think the default value of output_tolerance should be 0 and build_source should have an optional parameter to set it to something else, if necessary